src/coreclr/vm/prestub.cpp

```diff
@@ -1417,7 +1417,7 @@ namespace
 
         MethodTable* pMT = targetType.AsMethodTable();
 
-        CorElementType elemType = fieldType.GetSignatureCorElementType();
+        CorElementType elemType = fieldType.GetVerifierCorElementType();
```

Comment:
Is this going to allow mixing and matching the enum and the underlying type (e.g. the accessor can be `int` and the actual field can be `enum Something`)?

Reply:
Yes it will. Would you prefer I be more strict? To be fair that was the previous behavior as well.

Reply:
I think so. I'm not sure if that's desirable (it might be), but in any case we need to have tests which cover that, positive or negative.

Reply:
Do the method accessors work the same way?

Reply:
I believe they need to be exact since we are parsing method signatures. I will see about them being precise.

Reply:
Forget this approach. I need to compare signatures.